#28 feat: 마이페이지 API 및 UI 수정

odukong opened the pull request:
## 1. 무슨 이유로 코드를 변경했나요?

#28 를 구현했습니다.

## 2. 어떤 위험이나 장애를 발견했나요?

X

## 3. 관련 스크린샷을 첨부해주세요.

### 홈 UI 변경
![image](https://github.com/user-attachments/assets/cc13bebf-2aec-4146-9f36-7fd0d11be872)

### 마이페이지 사용자 정보 조회 API 
![스크린샷 2025-01-29 213519](https://github.com/user-attachments/assets/2795716d-6d97-41c6-baf8-6cbee6d61335)

### 마이페이지 받은+보낸 초대장 리스트 API (초대장 없는 상태)
![스크린샷 2025-01-29 213530](https://github.com/user-attachments/assets/79a1b703-9ccc-4f3a-901e-6c69d6655e49)

### 마이페이지 받은+보낸 초대장 리스트 API (초대장 있는 상태-임시데이터)
![스크린샷 2025-01-29 221230](https://github.com/user-attachments/assets/7ac9a9ae-1ce0-47b6-9837-a985082e89d3)
![스크린샷 2025-01-29 221241](https://github.com/user-attachments/assets/86cdfcc4-7346-4245-afba-1450b7190ead)

### 마이페이지 초대장 상세보기 API (임시)
![스크린샷 2025-01-29 213618](https://github.com/user-attachments/assets/5a4b34db-1923-4a36-8a26-c6c943bba242)


## 4. 완료 사항
- 마이페이지 API 연결 
  - 마이페이지 **사용자 정보 조회 API** 연결
  - 마이페이지 **받은, 보낸 초대장 리스트 정보 조회 API** 연결
  - 마이페이지 **초대장 삭제 API** 연결
  - 마이페이지 **초대장 상세조회 API** 연결 (> 참석, 불참석 API 구현 전으로 미연결상태)
  - 로그아웃 기능 구현
- 마이페이지UI 및 추가 UI 수정 완료
   -  초대장 삭제 모달 추가
   -  초대장 뒷면 **Info** 정렬 수정
   - lottie 파일 수정 및 홈화면 svg 파일 변경

## 5. 추가 사항
- 초대장 생성API 연결 이전, 참석자리스트 조회 API 구현 전인 상태로 초대장 리스트 조회 + 상세조회 API에 대한 테스트 진행이 덜 된 상태입니다. 추가적인 에러 발생 시, 참석/불참 리스트 API 연결하며 리팩토링도 함께 진행하겠습니다.
- **customAxios**의 응답 값 패러미터를 서버 응답값에 맞게 수정하였습니다. 참고부탁드리겠습니다!

odukong committed:
feat : 로그아웃 기능능 추가

ref : #28

odukong committed:
style : 홈 애니메이션 lottie 파일 수정

ref : #28

odukong committed:
style : 초대장 삭제 버튼 추가

ref : #28

odukong committed:
style : 템플릿 데이터 뒷면 컬러 수정

ref : #28

odukong committed:
refactor : AttendeeList 'title' params 추가

- 참석자 명단, 불참석자 명단에 AttendeeList 컴포넌트를 공통적으로 사용하기 위해 title params를 추가하여 재활용성을 높임.

ref : #28

odukong committed:
feat : 초대장 삭제 모달 추가

ref : #28

odukong committed:
feat : 마이페이지 사용자 정보조회 API

- 사용자 이름, 받은 초대장+보낸 초대장 개수를 받아오는 myPageInfoAPI를 구현하였습니다.

ref : #28

odukong committed:
feat : 받은, 보낸 초대장 리스트 정보 조회 API 구현

ref : #28

odukong committed:
feat : locale 데이터 날짜 변환 함수 추가

- YYYY.MM.DD 형식의 데이터 변환을 위한 formatDate 함수 구현

ref : #28

odukong committed:
feat : 초대장 삭제 API 구현

- 선택된 특정 id 초대장을삭제하는 API를 구현하였습니다.

ref : #28

odukong committed:
refactor : formatDate함수 util 분리 및 Invitation 타입 전역 선언

- 서버에서 받아오는 초대장 데이터 형식인 Invitation 타입을 전역타입으로 분리하였습니다.
- locale 데이터를 YYYY.MM.DD로 변환하는 formatDate 함수를 util함수로 분리하였습니다.

ref : #28

odukong committed:
feat : 초대장 상세조회 API 구현(임시구현)

- 참석자/불참석자 여부에 대한 API가 아직 구현되지 않아 임시data값으로 대체하여 상세조회 API를 구현하였습니다.
- 생성된 초대장 데이터가 존재하지 않아 테스트를 진행하기에 어려움이 있어 참석자/불참석자 여부 API에 대한 연결을 진행하며 리팩토링을 진행할 예정입니다.
- 초대장 UI를 InvitationCard 컴포넌트로 대체하여 사용하였습니다.

ref :#28

odukong committed:
style : 초대장 뒷면 IField 데이터 정렬

- 피그마 UI와 같이 Title과 Description은 중앙에 두되, Field 값은 왼쪽 정렬이 될 수 있도록 InvitationBack 컴포넌트 스타일 코드를 수정하였습니다.

ref : #28

odukong committed:
style : 홈 화면 로고파일 변경

- 홈 화면 로고를 수정된 로고 파일로 교체하였습니다.

ref : #28

yezzan9 commented on src/pages/mypage/InvitationDetail.tsx:
```diff
@@ -13,47 +21,56 @@ const data = {
+    remark: '와라',
+  });
+
+  useEffect(() => {
+    let temp = false; // (테스트용으로 삭제할 값입니다.)
+    if (temp && id) {
+      // 초대장 상세정보 API
+      invitationDetailAPI(resetUserInfo, showBoundary, id).then((res) => {
+        if (res.isSuccess && res.result !== null) {
+          setInvitation(res.result);
+        }
+      });
+    } else {
+      console.error('invitationId is missing');
+    }
+    // (참석자/불참석자 API) - 서버 미구현
+  }, []);
```
앗 안 그래도 이 부분 얘기하고 싶었는데..!!!
useEffect부분까지 묶어서 하나의 API hook으로 관리하면 API 호출과 그 외 UI / 로직을 분리할 수 있어서 가독성 면에서 좋을 것 같아요 ! 이게 무슨 얘기냐면 ,,,, 지금 QR 조회 연결은 다 했는데 CORS 에러때문에 아직 PR안 올렸던 거라,, PR 올릴테니 코드를 봐주시면 될 것 같아요🙈

odukong replied:
QR 조회 PR 확인하였습니다!

하나 궁금한 점이 `useEffect`와 함께 사용되는 api 호출을 hooks으로 분리하는 경우가 아닌,
ex. 버튼 클릭 시와 같은 특정 경우에만 호출되는 api도 `useXXX` hooks로 분리하는 편이 좋을까요?

yezzan9 replied:
음..🤔 함수명이 해당 함수의 동작을 명확하게 나타내도록 작성하여 (ex - deleteInvitation..??) 분리하는 편이 좋지 않을까요?!
분리하지 않더라도 onClick함수에 api 호출 함수를 직접적으로 작성하는 게 아니라 파일 내에서도 하나의 함수로 정의하여 해당 함수를 onClick함수로 전달하는 방식일 테니까 그냥 파일을 완전히 분리해도 좋을 것 같아요 !!

odukong replied:
오호 그렇네요! API 관련 코드에 대한 일관성 측면에서도 분리하는 편이 더 좋을 것 같아서 분리하는 방식으로 리팩토링 진행하겠습니당~!

yezzan9 commented on src/pages/mypage/InvitationDetail.tsx:
```diff
@@ -13,47 +21,56 @@ const data = {
-          <Text>{data.description}</Text>
-        </InvitationBack>
-      </Invitation>
+      <InvitationCard
+        title={invitation.title}
+        date={formatDate(invitation.schedule)}
+        location={invitation.location}
+        description={invitation.remark}
+        backgroundColor={template[data.templateKey].bg_color}
+        fontColor={template[data.templateKey].bg_text_color}
+      />
+      {/**카카오톡 공유(링크, QR) = isMine인 경우에만...*/}
+      <ShareList imgURL={data.img} />
```
해당 컴포넌트에 이미지 URL도 props로 전달받도록 수정 중이라 추후 수정이 필요할 것 같습니다!

odukong replied:
api hook 관련하여 코드 수정하며 **image url props**도 받을 수 있도록 함께 수정하겠습니다.!

yezzan9 commented on src/utils/customAxios.ts:
안그래도 저도 API 연결 중에 서버 응답값 중 response.data 값이 존재하지 않아서 고민하고 있었는데 역시 이 부분을 수정해야하군요💦

odukong replied:
이상한 점을 못 느끼다가 저도 console을 여러번 찍어보고 깨닫게 된 이슈였습니다 😃💦

yezzan9 commented:
혹시 이 경우는 디자인이 따로 없는 건가요?!
저희 공통 에러 컴포넌트처럼 디자인이 추가되면 좋을 것 같아요(디자이너님만 괜찮으시다면..🙈)
![image](https://github.com/user-attachments/assets/b1840c08-4de3-48ef-a0e1-87eb76094efa)

odukong commented:
> 혹시 이 경우는 디자인이 따로 없는 건가요?! 저희 공통 에러 컴포넌트처럼 디자인이 추가되면 좋을 것 같아요(디자이너님만 괜찮으시다면..🙈) 

디자인이 따로 있는 것 같진 않아서 임의로 구현해둔 상태입니다..! 일단 노션의 디자인작업제안란에 해당 케이스도 추가해두었습니다. 이 부분은 디자인만 수정하면 되는 거라 차후 순위로 미뤄도 괜찮을 것 같긴 합니다!

odukong committed:
refactor : 마이페이지 사용자정보 조회 api hook 분리

ref : #28

odukong committed:
refactor : 초대장리스트 API hooks 분리

- 받은, 보낸 초대장 리스트 조회 : useGetInvitationList
- 초대장 삭제 : useDeleteInvitation

ref : #28

odukong committed:
remove : 기존 API 파일 삭제

- use hooks 분리를 이용한 API 호출을 공통적으로 사용하기 위해 기존 api 파일을 삭제.

ref : #28

odukong committed:
refactor : InvitationCard imgUrl props 추가

- 초대장 공통 컴포넌트에 사용될 앞면 이미지를 위한 imgUrl props를 추가하였습니다.
- 초대장 상세조회는 참석/불참 API 구현 이후 함께 구현하도록 하겠습니다

ref : #28

odukong committed:
Merge branch 'main' of https://github.com/Leets-Official/Yes-FE into feat/#28/마이페이지API

odukong committed:
fix : InvitationCard imgUrl props 삭제(중복선언)

ref : #28

yezzan9 reviewed (approved):
와우 코드 라인수가 엄청 많이 줄었네요 !! 짱짱수고하셨습니다 ദ്ദി˶﻿ˆ꒳ˆ˵)

odukong commented:
> 와우 코드 라인수가 엄청 많이 줄었네요 !! 짱짱수고하셨습니다 ദ്ദി˶﻿ˆ꒳ˆ˵)

API hooks로 분리하도록하는 예진님의 굳 아이디어 덕분입니다(*˘︶˘*).｡.:*♡
